### PR TITLE
Feature lhs

### PR DESCRIPTION
**Pull request by timdmacdo**
LHS script. Rick gave me this file and I made some small changes and added a test function.

### CODE REVIEW

File: trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#
+# Created:  Jul 2016, R. Fenrich (outside of SUAVE code)
+# Modified: Mar 2017, T. MacDonald
+
+import numpy as np
+
+## If needed for mapping to normal distribution:
+#from scipy.stats.distributions import norm 
+
+def latin_hypercube_sampling(num_dimensions,num_samples,criterion=''):
```
Comment (planes) [inline trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py]: Can we add a name to the default criterion?

File: trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    
+    n = num_dimensions
+    samples = num_samples
+    
+    segsize = 1./samples
+    lhd = np.zeros((samples,n))
+    
+    if( criterion == "" ): # sample is randomly chosen from within segment
```
Comment (planes) [inline trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py]: I think these loops can be replaced by vector operations. How often are these scripts called? We can make this faster.

File: trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+# latin_hypercube_sampling.py
+#
+# Created:  Jul 2016, R. Fenrich (outside of SUAVE code)
+# Modified: Mar 2017, T. MacDonald
+
```
Comment (planes) [inline trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py]: Put an import header here.

File: trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py (hunk 1 above)
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py]: random?

File: trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py (hunk 2 above)
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py]: I imagine every time a new set of design variable samples needs to be created. The initial case would be for CFD calculation points so comparatively trivial. In general relative to other script I think pretty rarely but I can try changing it anyway.

File: trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py (hunk 1 above)
Comment (planes) [inline trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py]: Sure, that works. It's more descriptive than nothing.

File: trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py (hunk 2 above)
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py]: I replaced the nested loops. I didn't see an easy way to change shuffle loop (and the one I found online didn't work anyway). If you having any ideas that you think will still be easy to understand let me know.

Comment (planes) [inline trunk/SUAVE/Methods/Utilities/latin_hypercube_sampling.py]: That's good. I think I have an idea on how to get rid of that loop. I'm not as concerned about this one though.

**Comment by jmvegh** [comment]
Looks good

**Pull request merged**